packages/itwinui-css/src/table/base.scss

```diff
@@ -90,16 +92,23 @@
       opacity: 1;
     }
 
-    &:hover,
-    &:focus,
-    &:focus-within {
+    &:is(:hover, :focus-visible, :has(:focus-visible)) {
       background-color: var(--iui-color-background-hover);
 
-      .iui-table-sort,
-      .iui-table-filter-button {
+      :is(.iui-table-sort, .iui-table-filter-button) {
         opacity: 1;
       }
     }
+
+    @supports not selector(:has(+ *)) {
+      &:focus-within {
+        background-color: var(--iui-color-background-hover);
+
+        :is(.iui-table-sort, .iui-table-filter-button) {
+          opacity: 1;
+        }
+      }
+    }
```

Comment:
I read the second bullet point of the PR description but am still a little confused about why this change was needed. Could you elaborate?